contracts/pallet-ibc/src/light_clients.rs

```diff
@@ -232,21 +218,77 @@ pub enum AnyUpgradeOptions {
-#[derive(Clone, Debug, PartialEq, Eq, ClientState, Protobuf)]
-pub enum AnyClientState {
-	#[ibc(proto_url = "GRANDPA_CLIENT_STATE_TYPE_URL")]
-	Grandpa(ics10_grandpa::client_state::ClientState<HostFunctionsManager>),
-	#[ibc(proto_url = "BEEFY_CLIENT_STATE_TYPE_URL")]
-	Beefy(ics11_beefy::client_state::ClientState<HostFunctionsManager>),
-	#[ibc(proto_url = "TENDERMINT_CLIENT_STATE_TYPE_URL")]
-	Tendermint(ics07_tendermint::client_state::ClientState<HostFunctionsManager>),
-	#[ibc(proto_url = "WASM_CLIENT_STATE_TYPE_URL")]
-	Wasm(ics08_wasm::client_state::ClientState<AnyClient, Self, AnyConsensusState>),
-	#[cfg(test)]
-	#[ibc(proto_url = "MOCK_CLIENT_STATE_TYPE_URL")]
-	Mock(ibc::mock::client_state::MockClientState),
+mod v1 {
+	use super::*;
+
+	#[derive(Clone, Debug, PartialEq, Eq)]
+	pub enum AnyClientState {
+		Grandpa(ics10_grandpa::client_state::ClientStateV1<HostFunctionsManager>),
+		Beefy(ics11_beefy::client_state::ClientState<HostFunctionsManager>),
+		Tendermint(ics07_tendermint::client_state::ClientState<HostFunctionsManager>),
+		Wasm(ics08_wasm::client_state::ClientState<AnyClient, Self, AnyConsensusState>),
+		#[cfg(test)]
+		Mock(ibc::mock::client_state::MockClientState),
+	}
+
+	impl ibc::protobuf::Protobuf<Any> for AnyClientState {}
+
+	impl TryFrom<Any> for AnyClientState {
+		type Error = ics02_client::error::Error;
+		fn try_from(value: Any) -> Result<Self, Self::Error> {
+			match value.type_url.as_str() {
+				"" => Err(ics02_client::error::Error::empty_consensus_state_response()),
+				GRANDPA_CLIENT_STATE_TYPE_URL => Ok(Self::Grandpa(<ics10_grandpa::client_state::ClientStateV1<HostFunctionsManager>>::decode_vec(&value.value).map_err(::ibc::core::ics02_client::error::Error::decode_raw_client_state)?)),
+				BEEFY_CLIENT_STATE_TYPE_URL => Ok(Self::Beefy(<ics11_beefy::client_state::ClientState<HostFunctionsManager>>::decode_vec(&value.value).map_err(::ibc::core::ics02_client::error::Error::decode_raw_client_state)?)),
+				TENDERMINT_CLIENT_STATE_TYPE_URL => Ok(Self::Tendermint(<ics07_tendermint::client_state::ClientState<HostFunctionsManager>>::decode_vec(&value.value).map_err(::ibc::core::ics02_client::error::Error::decode_raw_client_state)?)),
+				WASM_CLIENT_STATE_TYPE_URL => Ok(Self::Wasm(<ics08_wasm::client_state::ClientState<AnyClient, Self, AnyConsensusState>>::decode_vec(&value.value).map_err(::ibc::core::ics02_client::error::Error::decode_raw_client_state)?)),
+				#[cfg(test)] MOCK_CLIENT_STATE_TYPE_URL => Ok(Self::Mock(<ibc::mock::client_state::MockClientState>::decode_vec(&value.value).map_err(::ibc::core::ics02_client::error::Error::decode_raw_client_state)?)),
+				_ => Err(ics02_client::error::Error::unknown_consensus_state_type(value.type_url)),
+			}
+		}
+	}
+
+	impl From<AnyClientState> for Any {
+		fn from(value: AnyClientState) -> Self {
+			match value {
+				AnyClientState::Grandpa(value) => Any {
+					type_url: GRANDPA_CLIENT_STATE_TYPE_URL.to_string(),
+					value: value.encode_vec().unwrap(),
```

Comment:
these unwraps are safe?

Reply:
Yes, the types are fixed and in the test in doesn't panic, so safe